akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -974,4 +974,29 @@ object Source {
       close: (S) => Future[Done]): Source[T, NotUsed] =
     Source.fromGraph(new UnfoldResourceSourceAsync(create, read, close))
 
+  /**
+   * Merge multiple [[Source]]s. Prefer the sources depending on the 'priority' parameters.
+   * The provided sources and priorities must have the same size and order.
+   *
+   * '''emits''' when one of the inputs has an element available, preferring inputs based on the 'priority' parameters if both have elements available
+   *
+   * '''backpressures''' when downstream backpressures
+   *
+   * '''completes''' when both upstreams complete (This behavior is changeable to completing when any upstream completes by setting `eagerComplete=true`.)
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def mergePrioritizedN[T](
+      sources: immutable.Seq[Source[T, _]],
+      priorities: immutable.Seq[Int],
+      eagerComplete: Boolean = false): Source[T, NotUsed] = {
```

Comment:
I wonder if `sourcesAndPriorities: immutable.Seq[(Source[T, _], Int))]` wouldn't be a better API than two seqs that need to be in sync, with a single seq there is no way you can accidentally mismatch the number of elements and it is also obvious which source the priority is for.